docs/fsharp/style-guide/formatting.md

```diff
@@ -1263,100 +1273,203 @@ module A2 =
     let function2 a b = a * a - b * b
 ```
 
-### Formatting object expressions and interfaces
+### Formatting do declarations
 
-Object expressions and interfaces should be aligned in the same way with `member` being indented after four spaces.
+In type declarations, module declarations and computation expressions, the use of `do` or `do!` is sometimes required for side-effecting operations.
+When these span multiple lines, use indentation and a new line to keep the indentation consistent with `let`/`let!`. Here is an example using `do` in a class:
 
 ```fsharp
-let comparer =
-    { new IComparer<string> with
-          member x.Compare(s1, s2) =
-              let rev (s: String) =
-                  new String (Array.rev (s.ToCharArray()))
-              let reversed = rev s1
-              reversed.CompareTo (rev s2) }
+// OK
+type Foo () =
+    let foo =
+        fooBarBaz
+        |> loremIpsumDolorSitAmet
+        |> theQuickBrownFoxJumpedOverTheLazyDog
+
+    do
+        fooBarBaz
+        |> loremIpsumDolorSitAmet
+        |> theQuickBrownFoxJumpedOverTheLazyDog
+
+// Bad - notice the "do" expression is indented one space less than the `let` expression
+type Foo () =
+    let foo =
+        fooBarBaz
+        |> loremIpsumDolorSitAmet
+        |> theQuickBrownFoxJumpedOverTheLazyDog
+    do fooBarBaz
+       |> loremIpsumDolorSitAmet
+       |> theQuickBrownFoxJumpedOverTheLazyDog
+```
+
+Here is an example with `do!` using 2 spaces of indentation (because with `do!` there is coincidentally no difference between the approaches when using 4 spaces of indentation):
+
+```fsharp
+// OK
+async {
+  let! foo =
+    fooBarBaz
+    |> loremIpsumDolorSitAmet
+    |> theQuickBrownFoxJumpedOverTheLazyDog
+    
+  do!
+    fooBarBaz
+    |> loremIpsumDolorSitAmet
+    |> theQuickBrownFoxJumpedOverTheLazyDog
+}
+
+// Bad - notice the "do!" expression is indented two spaces more than the `let!` expression
+async {
+  let! foo =
+    fooBarBaz
+    |> loremIpsumDolorSitAmet
+    |> theQuickBrownFoxJumpedOverTheLazyDog
+  do! fooBarBaz
+      |> loremIpsumDolorSitAmet
+      |> theQuickBrownFoxJumpedOverTheLazyDog
+}
 ```
 
-### Formatting white space in expressions
+## Formatting types and type annotations
+
```

Comment:
Maybe add a brief intro sentence here for the section?

Reply:
Done, thanks